_posts/2020-04-14-s3-access.md

```diff
@@ -0,0 +1,212 @@
+      "Resource": "arn:aws:::s3:mybucket/*"
+  }
+  ```
+
+  The above will match all items in the bucket. However, if you are giving bucket level permissions this won't work as the statement does not match "arn:aws:::s3:mybucket". Because of this, it's common to see the following in a resource policy.
+
+  ```json
+  {
+      "Resource": [
+          "arn:aws:::s3:mybucket",
+          "arn:aws:::s3:mybucket/*"
+      ]
+  }
+  ```
+
+- Action
+
+  Actions are the things we are allowed to do with resources. This will usually be things like PuObject, GetObject, ListBuckets - etc. Some actions work on buckets, whilst others work on the objects in those buckets.
+
+  It's usually reasonably obvious what is going on, except for the two things - "ListAllBuckets" and "ListBucket". The former lists all the buckets in the account, the latter lists all of the objects in a particular bucket. Yes, it's easy to mix up.
+
+- Effect
+
+  The effect is the result of various conditions matching the rest of the policy. We can do two things - we can either ALLOW things to happen or DENY them.
+
+- Principal
+
+  The principal is the identity of the 'thing' performing the action. For example, this lets us to block Mark from reading an object but allows Alice to read it. Users, roles, services, accounts, all of these things are 'principals'. Here are some examples.
+
+  Give access to the AWS Config service.
+
+  ```json
+  {
+
+      "Principal": {
+          "Service": "config.amazonaws.com"
+      }
+  }
+  ```
+
+  Give access to all accounts within AWS.
+  ```json
+  {
+      "Principal": {
+          "AWS": "*"
+      }
+  }
+  ```
+
+  Give access to two accounts - the following two examples are equivalent.
+  Important! Note the usage of 'root' doesn't mean root user, or root account. It just means the account is free to delegate permission as it sees fit.
+  ```json
+  {
+      "Principal": {
+         "AWS": [ "0123456789", "123456790" ]
+      }
+  }
+  ```
+
+  ```json
+  {
+      "Principal": {
+         "AWS": [ "arn:aws:iam::0123456789:root", "arn:aws:iam::123456790:root" ]
+      }
+  }
+  ```
+
+  Give access to everyone in the world
+  ```json
+  {
+      "Principal": "*"
+  }
+  ```
+
+  Give access to a specific role within an account
+  ```json
+  {
+      "Principal": {
+          "AWS": "arn:aws:iam::1234567890:role/my-cool-role"
+      }
+  }
+  ```
+
+  Note that the only time wildcards work in a policy is when they are specified as the only thing in a value - "*" will work, but "arn:aws:iam::1234567890:role/*" will not.
+
+- Condition
+
+  Condition statements allow us to further restrict access based on various circumstantial attributes. Maybe we block Mark all the time, except for when Mark is accessing the bucket from a particular office.
+
+- Canonical ID
+
+  Canonical ID's pop up in a few places but you are more likely to see them in ACLs than anywhere else. They usually identify what accounts may have access to certain buckets.
```

Comment:
May be worth providing some more information (like a link or a short paragraph) other than where a canonical id is likely to be found. eg, https://docs.aws.amazon.com/general/latest/gr/acct-identifiers.html